managing-os/templates/fedora/upgrade-26-to-27.md

```diff
@@ -0,0 +1,386 @@
+
+ 4. Trim the new template.
+
+        [user@fedora-27 ~]$ sudo fstrim -v /
+
+ 5. Shut down the new TemplateVM (from the command-line or Qubes VM Manager).
+
+        [user@dom0 ~]$ qvm-shutdown fedora-27
+
+ 6. Remove the cache file, if you created one.
+
+        [user@dom0 ~]$ rm /var/tmp/template-upgrade-cache.img
+
+ 7. (Recommended) Switch everything that was set to the old template to the new
+    template, e.g.:
+
+     1. Make the new template the default template:
+
+        Applications Menu --> System Tools --> Qubes Global Settings --> Default template
+
+     2. Base AppVMs on the new template. In Qubes Manager, for each VM that is
+        currently based on `fedora-26` that you would like to base on
+        `fedora-27`, enter its VM settings and change the Template selection:
+
+        Applications Menu --> (select a VM) --> VM settings --> Template
+
+     3. Base the [DispVM] template on the new template.
+
+            [user@dom0 ~]$ qvm-create -l red -t fedora-27 fedora-27-dvm
+            [user@dom0 ~]$ qvm-prefs fedora-27-dvm template_for_dispvms True
+            [user@dom0 ~]$ qvm-features fedora-27-dvm appmenus-dispvm 1
+            [user@dom0 ~]$ qubes-prefs default-dispvm fedora-27-dvm
+
+ 8. (Optional) Remove the old template. (Make sure to type `fedora-26`, not
+    `fedora-27`.)
+
+        [user@dom0 ~]$ sudo dnf remove qubes-template-fedora-26
+
+
+### Upgrading StandaloneVMs ###
+
+The procedure for upgrading a StandaloneVM from Fedora 26 to Fedora 27 is the
+same as for a TemplateVM.
+
+
+### Summary: Upgrading the Minimal Fedora 26 Template to Fedora 27 ###
+
+**Note:** The prompt on each line indicates where each command should be entered
+(`@dom0` or `@fedora-27`).
+
+        [user@dom0 ~]$ qvm-clone fedora-26-minimal fedora-27-minimal
+        [user@dom0 ~]$ qvm-run -u root -a fedora-27-minimal xterm
+        [root@fedora-27-minimal ~]# dnf clean all
+        [user@fedora-27-minimal ~]# dnf --releasever=27 --best --allowerasing distro-sync
+        [user@fedora-27-minimal ~]# fstrim -v /
+
+    (Shut down TemplateVM by any normal means.)
+
+(If you encounter insufficient space issues, you may need to use the methods
+described for the standard template above.)
+
+
+Additional Information
+----------------------
+
+As mentioned above, you may encounter the following `dnf` error:
+
+    At least X MB more space needed on the / filesystem.
+
+In this case, you have several options:
+
+ 1. [Increase the TemplateVM's disk image size][resize-disk-image].
+    This is the solution mentioned in the main instructions above.
+ 2. Delete files in order to free up space. One way to do this is by
+    uninstalling packages. You may then reinstalling them again after you
+    finish the upgrade process, if desired). However, you may end up having to
+    increase the disk image size anyway (see previous option).
+ 3. Increase the `root.img` size with `qvm-grow-root`.
```

Comment:
This duplicates the first point, which is more accurate (handle both 3.2 and 4.0).

Reply:
Thanks. Removing.